src/LDLFactorizations.jl

```diff
@@ -198,8 +193,8 @@ function ldl_numeric_upper!(n, Ap, Ai, Ax, Cp, Ci, Lp, parent, Lnz, Li, Lx, D, Y
       Lnz[i] += 1
       top += 1
     end
-    if dynamic_regul && abs(D[k]) < dynamic_dict[:tol]
-      r = P[k] <= dynamic_dict[:n_d] ? dynamic_dict[:r1] : dynamic_dict[:r2]
+    if dynamic_regul && abs(D[k]) < dynamic_args[:tol]
+      r = P[k] <= dynamic_args[:n_d] ? dynamic_args[:r1] : dynamic_args[:r2]
```

Comment:
Why not simply have keyword arguments `r1`, `r2` and `tol` with default value `zero(eltype(Ax))` and set
```julia
dynamic_reg = r1 > 0 || r2 > 0
```
`nd` could have default value `n`. It would make the function signature more explicit.

Reply:
Something like this? `n` is already used in `ldl_factorize!`.